.semaphore/semaphore.yml

```diff
@@ -19,60 +19,164 @@ queue:
   - when: "branch != 'main'"
     processing: parallel
 
-global_job_config:
-  prologue:
-    commands:
-      - checkout
-      - make ci-bin-sem-cache-restore
-      - |
-        if [[ "${SEMAPHORE_ORGANIZATION_URL}" == *".semaphoreci.com" ]]; then
-          echo "Skipping Vault setup for Semaphore CI"
-        else
-          . vault-setup
-        fi
-  epilogue:
-    always:
-      commands:
-        - make ci-bin-sem-cache-store
-        - make store-test-results-to-semaphore
```

Comment:
Mainly pushed into `Build & Test (VS Code: linux x64)` with YAML anchors for reuse on the non-Windows tasks